runtime/src/main/java/com/datastax/oss/quarkus/runtime/AbstractCassandraClientProducer.java

```diff
@@ -0,0 +1,120 @@
+import com.datastax.oss.driver.api.core.config.DriverConfig;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.ProgrammaticDriverConfigLoaderBuilder;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoader;
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultProgrammaticDriverConfigLoaderBuilder;
+import com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures;
+import com.datastax.oss.quarkus.config.CassandraClientConfig;
+import com.datastax.oss.quarkus.config.CassandraClientConnectionConfig;
+import com.typesafe.config.ConfigFactory;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.CompletionStage;
+
+public abstract class AbstractCassandraClientProducer {
+
+  private CassandraClientConfig config;
+
+  public void setCassandraClientConfig(CassandraClientConfig config) {
+    this.config = config;
+  }
+
+  private ProgrammaticDriverConfigLoaderBuilder createDriverConfigLoader() {
+    return new DefaultProgrammaticDriverConfigLoaderBuilder(
+        () -> {
+          ConfigFactory.invalidateCaches();
+          return ConfigFactory.defaultOverrides()
+              .withFallback(ConfigFactory.parseResources("application.conf"))
+              .withFallback(ConfigFactory.parseResources("application.json"))
+              .withFallback(ConfigFactory.defaultReference())
+              .resolve();
+        },
+        DefaultDriverConfigLoader.DEFAULT_ROOT_PATH) {
+      @NonNull
+      @Override
+      public DriverConfigLoader build() {
+        return new NonReloadableDriverConfigLoader(super.build());
+      }
+    };
+  }
+
+  CassandraClientConfig getCassandraClientConfig() {
+    return config;
+  }
+
+  public CqlSession createCassandraClient(CassandraClientConfig config) {
+    ProgrammaticDriverConfigLoaderBuilder configLoaderBuilder = createDriverConfigLoader();
+    configureConnectionSettings(configLoaderBuilder, config.cassandraClientConnectionConfig);
+    CqlSessionBuilder builder = CqlSession.builder().withConfigLoader(configLoaderBuilder.build());
+    return builder.build();
+  }
+
+  private void configureConnectionSettings(
+      ProgrammaticDriverConfigLoaderBuilder configLoaderBuilder,
+      CassandraClientConnectionConfig config) {
+    configLoaderBuilder.withStringList(DefaultDriverOption.CONTACT_POINTS, config.contactPoints);
+    configLoaderBuilder.withString(
+        DefaultDriverOption.LOAD_BALANCING_LOCAL_DATACENTER, config.localDatacenter);
+    config.requestTimeout.ifPresent(
+        v -> configLoaderBuilder.withDuration(DefaultDriverOption.REQUEST_TIMEOUT, v));
+    // force java clock because jnr does not work with quarkus
+    configLoaderBuilder.withBoolean(DefaultDriverOption.TIMESTAMP_GENERATOR_FORCE_JAVA_CLOCK, true);
+  }
```

Comment:
I believe there was some talk about this in the previous PR but this doesn't seem like a scalable configuration strategy at all.  We're forced into a double enumeration here; we have to specify the config props in one place and then remember to add them in another.  :unpleasant:

Probably something else we can address after we get this ticket in... but if there isn't a ticket for this already there probably should be.

Reply:
Yes, right now we decided to go with this approach:
- basic properties explicitly set using the quarkus config mechanism, all other driver settings could be passed using `application.conf` or `application.json`(see `CassandraClientConfigTest`).
There is also a task to improve it: https://datastax-oss.atlassian.net/browse/JAVA-2697